### PR TITLE
migrate to cucumber 2

### PR DESCRIPTION
**Pull request by cheahuychou**
Update `exo-` sub-projects to use cucumber 2 so that we can see useful error messages.

### CODE REVIEW

File: cli/features/support/env.ls [hunk 1]
```diff
@@ -1,22 +1,26 @@
...
-module.exports = ->
+defineSupportCode ({After, set-default-timeout, set-world-constructor}) ->
 
-  @set-default-timeout 2000
+  set-default-timeout 2000
+  set-world-constructor World
 
 
-  @After tags: ['~@e2e'], (scenario, done) ->
+  After tags: '~@e2e', (scenario, done) ->
```
Comment (charlierudolph) [inline cli/features/support/env.ls]: The tag syntax changed in cucumber 2.0. `~@e2e` should become `not @e2e`. Same for `~@docker-cleanup` below

File: exo-add/features/steps/steps-then.ls [hunk 2]
```diff
@@ -1,15 +1,17 @@
 require! {
   'chai' : {expect}
+  'cucumber': {defineSupportCode}
+
```
Comment (charlierudolph) [inline exo-add/features/steps/steps-then.ls]: please remove this newline

File: exo-clone/features/steps/steps-when.ls [hunk 3]
```diff
@@ -1,18 +1,20 @@
 require! {
   'chai' : {expect}
+  'cucumber': {defineSupportCode}
+
```
Comment (charlierudolph) [inline exo-clone/features/steps/steps-when.ls]: Please remove this newline

**Review by charlierudolph — COMMENTED** [review]
Please also remove the use of the `--strict` flag. This was removed in cucumber 2.0 as strict is now the default

**Review by charlierudolph — APPROVED** [review]
Thanks a lot for doing this! Good to ship once the comments are addressed if the build stays green

File: exo-clone/features/steps/steps-when.ls (hunk 3 above)
Comment (hugoesthere) [reply exo-clone/features/steps/steps-when.ls]: This comment was undressed

File: exo-sync/features/steps/steps-given.ls [hunk 4]
```diff
@@ -1,24 +1,25 @@
-require! {
+require! {
```
Comment (hugoesthere) [inline exo-sync/features/steps/steps-given.ls]: Remove trailing whitespace

**Review by hugoesthere — APPROVED** [review]
Awesome, thanks! 🚀 once tests are green

**Pull request closed without merging**